crates/router/src/connector/utils.rs

```diff
@@ -352,6 +370,102 @@ pub fn get_header_key_value<'a>(
         ))?
 }
 
+pub fn to_boolean(string: String) -> bool {
+    let str = string.as_str();
+    match str {
+        "true" => true,
+        "false" => false,
+        "yes" => true,
+        "no" => false,
+        _ => false,
+    }
+}
+
+pub fn get_connector_meta(
+    connector_meta: Option<serde_json::Value>,
+) -> Result<serde_json::Value, Error> {
+    connector_meta.ok_or_else(missing_field_err("connector_meta_data"))
+}
+
+pub fn to_connector_meta<T>(connector_meta: Option<serde_json::Value>) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let json = connector_meta.ok_or_else(missing_field_err("connector_meta_data"))?;
+    parse_struct(json)
+}
+
+pub fn to_connector_meta_from_secret<T>(
```

Comment:
Try to implement the function as chained function. Can be done in a separate PR too.